src/codecs/openexr.rs

```diff
@@ -152,10 +152,10 @@ impl<'a, R: 'a + Read + Seek> ImageDecoder<'a> for OpenExrDecoder<R> {
     }
 
     // reads with or without alpha, depending on `self.alpha_preference` and `self.alpha_present_in_file`
-    fn read_image_with_progress<F: Fn(Progress)>(
+    fn read_image_with_progress<F: FnMut(Progress)>(
```

Comment:
So, @HeroicKatora will there be any problem with recursion of the progress callback? If yes, is it even relevant?

Reply:
No, I don't believe so. Only a really small fraction will be recursive, if any, and our default implementation handles this gracefully.